xarray/core/datatree.py

```diff
@@ -1064,14 +1059,18 @@ def from_dict(
 
         # First create the root node
         root_data = d.pop("/", None)
-        obj = cls(name=name, data=root_data, parent=None, children=None)
+        if isinstance(root_data, DataTree):
+            obj = root_data.copy()
+            obj.orphan()
+        else:
+            obj = cls(name=name, data=root_data, parent=None, children=None)
 
         if d:
             # Populate tree with children determined from data_objects mapping
             for path, data in d.items():
                 # Create and set new node
                 node_name = NodePath(path).name
-                if isinstance(data, cls):
+                if isinstance(data, DataTree):
```

Comment:
So this was largely to make `mypy` happy, but I think it makes sense. Just using `cls`, I don't think `mypy` was realising that objects that were `DataTree` instances couldn't get to the `else` statement.